SignalMessaging/profiles/OWSProfileManager.m

```diff
@@ -718,62 +518,36 @@ - (void)clearProfileWhitelist
 {
     DDLogWarn(@"%@ Clearing the profile whitelist.", self.logTag);
 
-    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
-        @synchronized(self)
-        {
-            [self.userProfileWhitelistCache removeAllObjects];
-            [self.groupProfileWhitelistCache removeAllObjects];
-
-            [self.dbConnection purgeCollection:kOWSProfileManager_UserWhitelistCollection];
-            [self.dbConnection purgeCollection:kOWSProfileManager_GroupWhitelistCollection];
-            OWSAssert(0 == [self.dbConnection numberOfKeysInCollection:kOWSProfileManager_UserWhitelistCollection]);
-            OWSAssert(0 == [self.dbConnection numberOfKeysInCollection:kOWSProfileManager_GroupWhitelistCollection]);
-        }
-    });
+    [self.dbConnection asyncReadWriteWithBlock:^(YapDatabaseReadWriteTransaction *transaction) {
```

Comment:
Use one big transaction.   Also, I've started using async writes throughout the profile logic.